contrib/database/sql/conn_test.go

```diff
@@ -168,3 +170,47 @@ func TestWithChildSpansOnly(t *testing.T) {
 		})
 	}
 }
+
+func TestWithErrorCheck(t *testing.T) {
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	assertErrCheck := func(t *testing.T, mt mocktracer.Tracer, errExist bool, opts ...Option) {
+		Register("mysql", &mysql.MySQLDriver{})
+		defer unregister("mysql")
+
+		db, err := Open("mysql", "test:test@tcp(127.0.0.1:3306)/test", opts...)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer db.Close()
+
+		db.QueryContext(context.Background(), "SELECT a FROM "+tableName)
+
+		spans := mt.FinishedSpans()
+		assert.True(t, len(spans) > 0)
+
+		s := spans[len(spans)-1]
+		assert.Equal(t, errExist, s.Tag(ext.Error) != nil)
+	}
+
+	t.Run("defaults", func(t *testing.T) {
+		mt := mocktracer.Start()
```

Comment:
Why not include these in the function above?

Also, you could just do:
```go
t.Run("test-name", testOpts(WithErrorCheck(errFn)))
```
And define `testOpts` as:
```go
func testOpts(opts ...Option) func(t *testing.T) {
    return func(t *testing.T) {
        mt := mocktracer.Start()
        // ...
    }
}
```

Reply:
nice !